### PR TITLE
Step #1 of SDK refactoring

### PR DESCRIPTION
**Pull request by skidson**
- Made all model classes use getters and setters w/ private members and conventional member names
- Removed the ServiceNote class - this is stored in a vehicles key/value store and should be application specific (will move to Gauge)
- Did _not_ switch to Retrofit (considering this may be more suitable when we switch to v2 APIs, will re-evaluate then)
- Changed network stack to use OKHttp (more performant)
- Upgraded Gson version
- Removed unused resources (drawables, styles, etc)

Note: this is very backwards incompatible - I've created a release/czech branch in this repo too so we can hotfix safely

### CODE REVIEW

File: MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/MojioClient.java [hunk 1]
```diff
@@ -257,30 +239,7 @@ public void onFailure(ResponseError error) {
...
      * @param responseListener
      */
-    public void login(final String userNameOrEmail, final String password, final ResponseListener<User> responseListener) {
-
-        // If we need an app access token, then fetch it first and call login again.
-        // We will need an app access token under the following conditions:
-        // 1. There is no access token at all (caught by the shouldRefreshAccessToken() call)
-        // 2. The access token needs to be refreshed (caught by the shouldRefreshAccessToken() call)
-        // 3. The access token is not an app access token
-        if (_oauthHelper.shouldRefreshAccessToken() || _oauthHelper.isUserToken()) {
```
Comment (MauroHorie) [inline MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/MojioClient.java:267]: Just want to double-check if this is no longer necessary.

File: MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/MojioClient.java [hunk 2]
```diff
@@ -291,18 +250,13 @@ public void onFailure(ResponseError error) {
         }
 
         String entityPath = String.format("Login/%s?secretKey=%s&userOrEmail=%s&password=%s&minutes=%d",
```
Comment (MauroHorie) [inline MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/MojioClient.java:269]: Maybe we should not proceed if the encoding is invalid as the username and/or password will be null.

File: MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/models/Observers/DiagnosticCodeObserver.java [hunk 3]
```diff
@@ -1,13 +1,30 @@
...
 
 /**
+ * Model class for a diagnostic code observer.
  * Created by jian on 01/04/2015.
  */
 public class DiagnosticCodeObserver extends ConditionalObserverBase{
 
-    public ArrayList<Object> DiagnosticCodes;
+    @SerializedName("DiagnosticCodes")
+    private ArrayList<Object> diagnosticCodes;
```
Comment (MauroHorie) [inline MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/models/Observers/DiagnosticCodeObserver.java:16]: Maybe add a TODO to see if we can transform the Object into an actual type.

File: MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/models/Observers/EventObserver.java [hunk 4]
```diff
@@ -1,11 +1,30 @@
...
 
 import java.util.ArrayList;
 
 /**
+ * Model class for an event observer.
  * Created by jian on 01/04/2015.
  */
 public class EventObserver extends ConditionalObserverBase{
 
-    public ArrayList<Object> EventTypes;
```
Comment (MauroHorie) [inline MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/models/Observers/EventObserver.java:10]: Same here for Object.

File: MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/models/Observers/ObservableAttribute.java [hunk 5]
```diff
@@ -1,10 +1,40 @@
...
  */
 public class ObservableAttribute {
 
-    public String LinkType;
-    public Object TypeId;
+    @SerializedName("LinkType")
+    private String linkType;
+
+    @SerializedName("TypeId")
+    private Object typeId;
```
Comment (MauroHorie) [inline MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/models/Observers/ObservableAttribute.java:16]: Same here for Object.

File: MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/networking/MojioHttpStack.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    public MojioHttpStack() {
+        this(new OkUrlFactory(new OkHttpClient()));
+    }
+
+    public MojioHttpStack(@NonNull OkUrlFactory okUrlFactory) {
+        this.okUrlFactory = okUrlFactory;
+    }
+
+    @Override protected HttpURLConnection createConnection(URL url) throws IOException {
+        return okUrlFactory.open(url);
```
Comment (MauroHorie) [inline MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/networking/MojioHttpStack.java:45]: Is this smart enough to open an HttpsURLConnection (note the s) if possible?

File: MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/networking/MojioRequest.java [hunk 7]
```diff
@@ -49,16 +49,15 @@
     private static final String TAG = MojioRequest.class.getSimpleName();
     private static final String PROTOCOL_CHARSET = "utf-8";
     private static final String PROTOCOL_CONTENT_TYPE = String.format("application/json; charset=%s", PROTOCOL_CHARSET);
+    private static final Gson GSON = new Gson();
 
     private Context mAppContext;
```
Comment (MauroHorie) [inline MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/networking/MojioRequest.java:54]: I just know you want to rename this :)

**Comment by MauroHorie** [comment]
This is very good. I really like the changes. The code is a lot cleaner and more correct from an engineering point of view. Unfortunately, the addition of getters and setters makes it not backwards compatible. Let's put this in a separate dev branch.

File: MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/MojioClient.java (hunk 1 above)
Comment (skidson) [inline MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/MojioClient.java:267]: It is, bad merge. Very nice catch :)

File: MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/MojioClient.java (hunk 2 above)
Comment (skidson) [inline MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/MojioClient.java:269]: if the exception gets thrown it'll just use the unencoded username/password. UnsupportedEncodingException really isn't something that'll ever get thrown but, if for some reason it does, the call will only fail if the user's username or password contains a symbol that must be encoded so there's a chance we're still OK.

In reality we should never pass login information as a URL parameter - I'm going to open a TP bug to have this looked at as it is definitely a security issue.

Comment (skidson) [inline MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/MojioClient.java:269]: https://mojio.tpondemand.com/entity/5857

File: MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/models/Observers/DiagnosticCodeObserver.java (hunk 3 above)
Comment (skidson) [inline MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/models/Observers/DiagnosticCodeObserver.java:16]: Good call, added a TODO. I found some instances of this sort of thing in a few places but the server sometimes expected a String of comma-separated values (meaning we'd need a custom Gson serialization strategy). We'll have to sweep through and verify the server's expectations for each of these so I've mentioned that in the TODO as well

File: MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/networking/MojioHttpStack.java (hunk 6 above)
Comment (skidson) [inline MojioSDK/mojiosdksrc/src/main/java/io/moj/mobile/android/sdk/networking/MojioHttpStack.java:45]: https://github.com/square/okhttp/blob/master/okhttp-urlconnection/src/main/java/com/squareup/okhttp/OkUrlFactory.java#L57

Yep :)

**Comment by skidson** [comment]
Good call - I'll push these changes to a "dev_refactored" branch

**Pull request closed without merging**